smartsim/_core/entrypoints/dragon_install.py

```diff
@@ -0,0 +1,309 @@
+
+from smartsim.error.errors import SmartSimCLIActionCancelled
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def check_for_utility(util_name: str) -> str:
+    """Check for existence of the provided CLI utility.
+
+    :param util_name: CLI utility to locate
+    :type util_name: str
+    :returns: Full path to executable if found. Otherwise, empty string
+    :rtype: str"""
+    utility = shutil.which(util_name)
+    if not utility:
+        logger.debug(f"{util_name} not available for Cray EX platform check.")
+    return utility or ""
+
+
+def _execute_platform_cmd(cmd: str) -> t.Tuple[str, int]:
+    """Execute the platform check command as a subprocess
+
+    :param cmd: the command to execute
+    :type cmd: str:
+    :returns: True if platform is cray ex, False otherwise
+    :rtype: bool"""
+    with subprocess.Popen(
+        cmd.split(),
+        stdout=subprocess.PIPE,
+    ) as popen:
+        out, _ = popen.communicate()
+        return out.decode("utf-8"), popen.returncode
+
+
+def is_crayex_platform() -> bool:
+    """Returns True if the current platform is identified as Cray EX and
+    HSTA-aware dragon package can be installed, False otherwise.
+
+    :returns: True if current platform is Cray EX, False otherwise
+    :rtype: bool"""
+
+    # ldconfig -p | grep cray | grep pmi.so &&
+    # ldconfig -p | grep cray | grep pmi2.so &&
+    # fi_info | grep cxi\
+    ldconfig = check_for_utility("ldconfig")
+    fi_info = check_for_utility("fi_info")
+    if not all((ldconfig, fi_info)):
+        logger.warning("Unable to validate Cray EX platform. Aborting dragon install")
+        return False
+
+    locate_msg = "Unable to locate %s. Aborting dragon install"
+
+    ldconfig1 = f"{ldconfig} -p"
+    ldc_out1, _ = _execute_platform_cmd(ldconfig1)
+    target = "pmi.so"
+    candidates = [x for x in ldc_out1.split("\n") if "cray" in x]
+    pmi1 = any(x for x in candidates if target in x)
+    if not pmi1:
+        logger.warning(locate_msg, target)
+        return False
+
+    ldconfig2 = f"{ldconfig} -p"
+    ldc_out2, _ = _execute_platform_cmd(ldconfig2)
+    target = "pmi2.so"
+    candidates = [x for x in ldc_out2.split("\n") if "cray" in x]
+    pmi2 = any(x for x in candidates if target in x)
+    if not pmi2:
+        logger.warning(locate_msg, target)
+        return False
+
+    fi_info_out, _ = _execute_platform_cmd(fi_info)
+    target = "cxi"
+    cxi = any(x for x in fi_info_out.split("\n") if target in x)
+    if not cxi:
+        logger.warning(locate_msg, target)
+        return False
+
+    return True
+
+
+def python_version() -> str:
+    """Return a formatted string used to filter release assets
+    for the current python version"""
+    return f"py{sys.version_info.major}.{sys.version_info.minor}"
+
+
+def dragon_pin() -> str:
+    """Return a string indicating the pinned major/minor version of the dragon
+    package to install"""
+    return "dragon-0.8"
```

Comment:
**NOTE**: we'll have to change this when we merge and know what version to pin but this version works. I avoided adding to `config.py` for the purpose of keeping the dragon installation code in one place, but am open to making it user-replaceable (e.g. `return os.environ['dragon_pin']`)